aws-xray-recorder-sdk-sql/src/main/java/com/amazonaws/xray/sql/ConnectionInfo.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amazonaws.xray.sql;
+
+import java.util.Objects;
+import org.checkerframework.checker.nullness.qual.Nullable;
+
+final class ConnectionInfo {
+    @Nullable
+    private final String sanitizedUrl;
+    @Nullable
+    private final String user;
+    @Nullable
+    private final String host;
+    @Nullable
+    private final String dbName;
+
+    private ConnectionInfo(@Nullable String sanitizedUrl,
+                           @Nullable String user,
+                           @Nullable String host,
+                           @Nullable String dbName) {
+        this.sanitizedUrl = sanitizedUrl;
+        this.user = user;
+        this.host = host;
+        this.dbName = dbName;
+    }
+
+    String getSanitizedUrl() {
+        return sanitizedUrl;
+    }
+
+    String getUser() {
+        return user;
+    }
+
+    String getHost() {
+        return host;
+    }
+
+    String getDbName() {
+        return dbName;
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+        ConnectionInfo that = (ConnectionInfo) o;
+        return Objects.equals(sanitizedUrl, that.sanitizedUrl) &&
+            Objects.equals(user, that.user) &&
+            Objects.equals(host, that.host) &&
+            Objects.equals(dbName, that.dbName);
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hashCode(sanitizedUrl);
```

Comment:
Is this sufficient to only hash the sanitized URL? Originally I hashed all 4 fields, but checkStyle called me out for using `hash` instead of `hashCode`.

Reply:
Did you use `Objects.hash(sanitizedUrl, user, host, dbName)` (varargs)? That's pretty weird. Maybe you can use IntelliJ to generate equals / hashcode so it just uses multiply / add directly

Reply:
Yeah, `Objects.hash()` is actually what IntelliJ generate used by default.

Reply:
I think you can use "IntelliJ default" to get the multiply version

Reply:
Ohhh I had never seen that before, thanks for the tip!